pkg/controller/job/job_controller.go

```diff
@@ -846,6 +840,35 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 			jobCtx.podsWithDelayedDeletionPerIndex = getPodsWithDelayedDeletionPerIndex(logger, jobCtx)
 		}
 	}
+
+	complete := false
+	if jobCtx.finishedCondition == nil {
+		if job.Spec.Completions == nil {
+			// This type of job is complete when any pod exits with success.
+			// Each pod is capable of
+			// determining whether or not the entire Job is done.  Subsequent pods are
+			// not expected to fail, but if they do, the failure is ignored.  Once any
+			// pod succeeds, the controller waits for remaining pods to finish, and
+			// then the job is complete.
+			complete = jobCtx.succeeded > 0 && active == 0
+		} else {
+			// Job specifies a number of completions.  This type of job signals
+			// success by having that number of successes.  Since we do not
+			// start more pods than there are remaining completions, there should
+			// not be any remaining active pods once this count is reached.
+			complete = jobCtx.succeeded >= *job.Spec.Completions && active == 0
```

Comment:
Here, the `complete` is now determined based on `active` before calling `manageJob`, while before it was computed after calling the `manageJob`. 

IIUC the scenario ( `active > 0` before `manageJob`, and `active == 0` after `manageJob`) is possible on job scaledown (reducing parallelism and completions). 

So, I'm wondering if we can fix the issue without affecting that scenario. If this is more complicated than I'm leaning to accept the fix, because the delay for marking the job as complete does not seem signifficant, but @alculquicondor for visibility.

Reply:
I found that some other problems will happen once we move the `manageJob` performing to before L846. Because `manageJob` will perform operations for the creation and deletion pods.

So, I will try to perform computing `active` before L846 for the scaledown scenario.

Reply:
@mimowo I improved computing `active` logic for the scaledown scenario.
Could you take another look?

Reply:
sgtm, reponded in the other related thread. Feel free to resolve this one